__integrations__/utils/cosmosdb_model.test.ts

```diff
@@ -149,6 +155,61 @@ describe("find", () => {
   });
 });
 
+describe("findAllVersionsBySearchKey", () => {
+  class MyVersionedModel extends CosmosdbModelVersionedTTL<
+    MyDocument,
+    NewMyDocument,
+    RetrievedMyVersionedDocument,
+    "pk"
+  > {
+    constructor(c: Container) {
+      super(c, NewMyDocument, RetrievedMyVersionedDocument, "pk");
+    }
+  }
+
+  const RetrievedMyVersionedDocument = t.intersection([
+    MyDocument,
+    RetrievedVersionedModel
+  ]);
+
+  type RetrievedMyVersionedDocument = t.TypeOf<
+    typeof RetrievedMyVersionedDocument
+  >;
+
+  it("should return all documents belonging to the same partition key", async () => {
+    const context = createContext("id");
+    await context.init();
+    const model = new MyVersionedModel(context.container);
+
+    await model.upsert({
+      id: testId,
+      pk: testPartition,
+      test: "test"
+    })();
+
+    await model.upsert({
+      id: "2testId" as NonEmptyString,
+      pk: testPartition,
+      test: "test"
+    })();
+
+    await model.upsert({
+      id: "3testId" as NonEmptyString,
+      pk: "invalidPartition",
+      test: "test"
+    })();
+
+    const result = await model.findAllVersionsBySearchKey([testPartition])();
+
+    expect(E.isRight(result)).toBeTruthy();
+    if (E.isRight(result)) {
+      // expect(O.isNone(result.right)).toBeTruthy();
```

Comment:
ditto

Reply:
Removed in [4f31c4d](https://github.com/pagopa/io-functions-commons/pull/296/commits/4f31c4da58599e82e5ba5769adbbe0dd0a1affca)